src/main/thrift/hive_metastore.thrift

```diff
@@ -1759,6 +1776,9 @@ service ThriftHiveMetastore extends fb303.FacebookService
   // alter table not only applies to future partitions but also cascade to existing partitions
   void alter_table_with_cascade(1:string dbname, 2:string tbl_name, 3:Table new_tbl, 4:bool cascade)
                        throws (1:InvalidOperationException o1, 2:MetaException o2)
+  AlterTableResponse alter_table_req(1:AlterTableRequest req)
```

Comment:
`c8f4984fe934af0fb65282bb6218b3baa6c2ca08` appears in hive `master` after 3.x line.
Was it backported to Hive 3.x and/or Hive 2.x?

Reply:
`alter_table_req` doesn't exist in https://github.com/apache/hive/blob/rel/release-3.1.2/standalone-metastore/src/main/thrift/hive_metastore.thrift

It's used in HDP3, so it seems like we should also use it when implementing writes. Of course, this leads to the question as to how ACID works for open source Hive 3.x. This is difficult to answer since none of us have been able to get it running.

Reply:
Thanks for explaining. Makes sense. 

> question as to how ACID works for open source Hive 3.x

It matters only if there is sufficient number of open source Hive 3.x users.
Do we know?

I'd guess HDP 3 may be a more popular version than open source Hive 3.x.
Later, we can handle multiple versions with the "alternative call" trick.

Reply:
Based on my experience of trying to run it, I might wager to guess zero users, so it's more of a curiosity than a question of practical concern.